### ISSUE TITLE
Build: Set default build type to profile

### ISSUE THREAD

**Issue by fabioanderegg** [o3de/o3de#2527]
**Is your feature request related to a problem? Please describe.**
In Visual Studio (and probably in other cmake generated build files) the `debug` build type is the default. I missed to change that to profile quite a few times, which is especially painful when building the whole engine. Also, it seems like there are still issues with running the Asset Processor when built in debug mode, and the performance of the Editor is very bad.

**Describe the solution you'd like**
Set the default cmake/Visual Studio build type to `profile`.

**Comment by amznestebanpapp** [comment]
Sadly, this looks like something we wont be able to change. I also have hit the "building debug unintentionally".

The default build type is not based on the order that they have in the sln file (which is what changes if we change the order of `CMAKE_CONFIGURATION_TYPES`). Therefore, changing the order of `CMAKE_CONFIGURATION_TYPES` does not affect it.

Found in multiple places that VS will always pick "debug" as the default one, no matter what order you pass. However, if you don't have any "debug" configuration, it will be alphabetically-based. I was able to confirm with the following tests:
- configs: debug, profile, release -> picks debug
- configs: profile, debug, release -> picks debug
- configs: profile, release -> picks profile
- configs: release, profile -> picks profile
- configs 0_profile, 1_release -> picks 0_profile
- configs 0_release, 1_profile -> picks 0_release
- configs 1_profile, 0_release -> picks 0_release

Since this is specific to VS and it would require us to change how we name configs, I dont think we should change it. Instead, we should put a feature request to Visual Studio to have a parameter in the `sln` file that defines the default configuration.
